x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+export interface CreatedPipeline {
+  created: string;
+}
+
+// TODO: document
+export const createMlInferencePipeline = async (
+  pipelineName: string,
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<CreatedPipeline> => {
+  const inferencePipelineGeneratedName = `ml-inference-${pipelineName}`;
+
+  // Check for existing pipeline
+  let exists = false;
+  try {
+    await esClient.ingest.getPipeline({ id: inferencePipelineGeneratedName });
+    exists = true;
+  } catch (err) {
+    // NOP
+  }
+  if (exists) {
+    throw new Error(`Pipeline ${inferencePipelineGeneratedName} already exists`);
```

Comment:
Currently this results in an unhandled error on the client side and the message is not visible:
```
{
    "statusCode": 502,
    "error": "Bad Gateway",
    "message": "Enterprise Search encountered an error. Check Kibana Server logs for details.",
    "attributes": {
        "error_code": "uncaught_exception"
    }
}
```

Thoughts on how we can bubble this message up?

Reply:
Would it be possible instead of a `throw` here, to return a 400 / Bad Request, and have the response body with the error. I'm not sure off the top of my head, but I'm sure there is a standard response / precedent that can be used (e.g. something like - and just an example - `{ "created": "false", "errors": ["Pipeline abcdefg already exists"] }`)